pdsClient/pdsClient.go

```diff
@@ -185,6 +186,19 @@ func (c *E3dbPDSClient) InternalModifiedSearch(ctx context.Context, params Inter
 	return result, e3dbClients.FlatMapInternalError(*internalError)
 }
 
+// InternalSearchAllowedReads returns the allowed reads that match the
+// provided params, returning a paginated list of allowed reads and error (if any).
+func (c *E3dbPDSClient) InternalSearchAllowedReads(ctx context.Context, params InternalSearchAllowedReadsRequest) (*InternalSearchAllowedReadsResponse, error) {
+	var result *InternalSearchAllowedReadsResponse
+	path := c.Host + "/internal/" + PDSServiceBasePath + "/allowed_reads"
+	request, err := e3dbClients.CreateRequest("POST", path, params)
+	if err != nil {
+		return result, err
+	}
+	internalError := e3dbClients.MakeE3DBServiceCall(c.E3dbAuthClient, ctx, request, &result)
+	return result, e3dbClients.FlatMapInternalError(*internalError)
```

Comment:
As this `FlatMap` function continues to show up more often, the more I want to update this code base to remove this custom error type from direct returns...

Reply:
Agree, I think we can keep the usefulness of the custom error by having the `MakeE3DBServcieCall` function return type error(returning nil if no error, and the RequestError struct otherwise)  and having callers of these methods use runtime type assertion to cast it to a ResponseError as needed/desired.